src/gui/DatabaseWidget.cpp

```diff
@@ -524,8 +524,10 @@ void DatabaseWidget::deleteEntries(QList<Entry*> selectedEntries)
     bool permanent = (recycleBin && recycleBin->findEntryByUuid(selectedEntries.first()->uuid()))
                      || !m_db->metadata()->recycleBinEnabled();
 
-    if (!confirmDeleteEntries(selectedEntries, permanent)) {
-        return;
+    if (!config()->get(Config::Security_DeleteEntryNoConfirm).toBool()) {
+        if (!confirmDeleteEntries(selectedEntries, permanent)) {
```

Comment:
The config check should be in the `confirmDeleteEntries` function. If the setting is true than confirmation is implied. If the recycle bin is not enabled (this is a database specific setting) than the confirmation dialog should ALWAYS be shown.

Reply:
Agreed ! 

Considering the actual wording of the checkbox: should we change it to something like "Delete entries to recycle bin without confirmation dialog" to make it more clear what is the consequence of ticking it ?

Reply:
Recommend: Move entries to recycle bin without confirmation

Reply:
Basically permanent deletion should always be confirmed, merely moving to the bin can be toggled. The default setting should be true (move without confirmation).